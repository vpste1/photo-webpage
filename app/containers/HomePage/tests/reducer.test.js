
import homePageReducer from '../reducer';

import {
  defaultAction,
  pageLoadAction,
  retrieveURLsSuccess,
  retrieveURLsFailure } from '../actions';

describe('Container [App] - reducers', () => {
  let state;
  beforeEach(() => {
  });
  it('should return the initial state', () => {
    expect(homePageReducer(undefined, {})).toMatchSnapshot();
  });
  it('should handle the defaultAction action correctly', () => {
    expect(homePageReducer(state, defaultAction())).toMatchSnapshot();
  });
  it('should handle the pageLoadAction action correctly', () => {
    expect(homePageReducer(state, pageLoadAction())).toMatchSnapshot();
  });
  it('should handle the retrieveURLsSuccess action correctly', () => {
    expect(homePageReducer(state, retrieveURLsSuccess())).toMatchSnapshot();
  });
  it('should handle the retrieveURLsFailure action correctly', () => {
    expect(homePageReducer(state, retrieveURLsFailure())).toMatchSnapshot();
  });
});
