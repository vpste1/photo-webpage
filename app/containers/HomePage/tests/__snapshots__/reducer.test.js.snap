// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container [App] - reducers should handle the defaultAction action correctly 1`] = `
Immutable.Map {
  activePage: null,
  dataRetrieved: false,
  titleList: Immutable.List [
  ],
  urlList: Immutable.List [
  ],
}
`;

exports[`Container [App] - reducers should handle the pageLoadAction action correctly 1`] = `
Immutable.Map {
  activePage: null,
  dataRetrieved: false,
  titleList: Immutable.List [
  ],
  urlList: Immutable.List [
  ],
}
`;

exports[`Container [App] - reducers should handle the retrieveURLsFailure action correctly 1`] = `
Immutable.Map {
  activePage: null,
  dataRetrieved: false,
  titleList: Immutable.List [
  ],
  urlList: Immutable.List [
  ],
}
`;

exports[`Container [App] - reducers should handle the retrieveURLsSuccess action correctly 1`] = `
Immutable.Map {
  activePage: null,
  dataRetrieved: true,
  titleList: undefined,
  urlList: undefined,
}
`;

exports[`Container [App] - reducers should return the initial state 1`] = `
Immutable.Map {
  activePage: null,
  dataRetrieved: false,
  titleList: Immutable.List [
  ],
  urlList: Immutable.List [
  ],
}
`;
